### PR TITLE
#31 WEb scraping for RJPS

### PR DESCRIPTION
**Pull request by idisblueflash**

### CODE REVIEW

File: scraping/spiders/poland_rjps.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    def _build_urls(self, file_name: str) -> List[str]:
+        ids = open_json_file(file_name)
+        return [f'{DETAIL_BASE_URL}={value["id"]}' for value in ids]
+
+    def parse(self, response, category: str):
+        import pdb;
+        pdb.set_trace()
+        raise NotImplemented
+
```
Comment (littlepea) [inline scraping/spiders/poland_rjps.py:42]: Don't forget unit tests

File: tests/spiders/test_poland_rjps.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        description = QuotesSpider().parse(normal_place, category='test').get('description')
+        assert 'tel. 85 7188100' in description
+
+    def test_parse_website(self, normal_place):
+        description = QuotesSpider().parse(normal_place, category='test').get('description')
+        assert 'http://bip.mops.um.zabludow.wrotapodlasia.pl' in description
+
+    def test_parse_update_date(self, normal_place):
+        description = QuotesSpider().parse(normal_place, category='test').get('description')
+        assert 'data aktualizacji:  2018-02-05' in description
```
Comment (idisblueflash) [inline tests/spiders/test_poland_rjps.py:223]: # TODO 

it's better to format the update date as `data updated: 2018-02-05`

Comment (idisblueflash) [reply tests/spiders/test_poland_rjps.py]: https://github.com/littlepea/safeway-data/pull/18/commits/fd705666c5da0a19151a26bd5263698b63437353

File: scraping/spiders/poland_rjps.py (hunk 1 above)
Comment (idisblueflash) [reply scraping/spiders/poland_rjps.py:98]: https://github.com/littlepea/safeway-data/pull/18/files#diff-0c292a0576fea8a6ab9743892f264e458a499464dfc87b3c41050553d0de9c58

**Comment by idisblueflash** [comment]
@littlepea please help to review

**Comment by idisblueflash** [comment]
### How to fetch ids from a category
1. open [the category page](https://rjps.mpips.gov.pl/RJPS/WJ/start.do?podkategorie=35,36,37,38,39,40&stronaListy=1&liczbaPozycjiLista=20&wersja=1&widocznyPanel=lista) in your browser
2. get payload from the different category
```md
| Category                      | Category IDs      |
|-------------------------------+-------------------|
| Seniors                       | 21,22,23,24,20,19 |
| Family                        | 16,13,17,18,14,15 |
| People  with disablilities    | 7,8,9,10,11,12    |
| Children and youth            | 3,4,1,2,5,6       |
| Addiction                     | 31,32,33,34       |
| Health                        | 25,26,27,28,29,30 |
| Violence                      | 41,45,42,43,44    |
| Difficult financial situation | 35,36,37,38,39,40 |
```
4. fetch ids response with Postman or script
  1. select ids from step 2., for example, `21,22,23,24,20,19`
  2. run script to fetch response
  ```sh
curl --location --request POST 'https://rjps.mpips.gov.pl/RJPS/WJ/wyszukiwanie/zaladujDane.do' \
--header 'Content-Type: application/json' \
--data-raw '{
    "filtry": {
        "miejscowosc": "",
        "wojewodztwo": "",
        "powiat": "",
        "gmina": "",
        "odleglosc": 0,
        "lokalizacja": "",
        "szukanaFraza": "",
        "podkategorie": "21,22,23,24,20,19",
        "stronaListy": 1,
        "liczbaPozycjiLista": "20",
        "wersja": "1",
        "widocznyPanel": "mapa",
        "wejscieKarta": "",
        "idJednostki": ""
    },
    "stronaListy": 1
}'
```

File: data/spiders/rips/addiction.json [hunk 3]
```diff
@@ -0,0 +1,9232 @@
+[
```
Comment (littlepea) [inline data/spiders/rips/addiction.json:1]: Why do you need to store these files in git? Shouldn't your spider fetch them dynamically?

File: data/spiders/rips/missing_info_page.html [hunk 4]
```diff
@@ -0,0 +1,154 @@
+<html>
+<head></head>
```
Comment (littlepea) [inline data/spiders/rips/missing_info_page.html:2]: Why do you need to store HTML in git?

File: scraping/spiders/poland_rjps.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+import json
+
+from joblib import Memory
+from config.settings import Settings
+
+memory = Memory(location='cache/poland_rjps')
+settings = Settings()
+DATA_PATH = f'{settings.spider_data_path}/rips'
+COUNTRY_NAME = 'Poland'
+DEFAULT_CATEGORY = "Any Help"
```
Comment (littlepea) [inline scraping/spiders/poland_rjps.py:14]: Should we move `DEFAULT_CATEGORY` to `config.constants`?

File: scraping/spiders/poland_rjps.py [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+CATEGORY_MAPPING = {
+    'Medical': ['seniors.json', 'health.json'],
```
Comment (littlepea) [inline scraping/spiders/poland_rjps.py:17]: I think these JSONs should be scraped dynamically too instead of coming from static files in `DATA_PATH`

**Review by littlepea — CHANGES_REQUESTED** [review]

File: data/spiders/rips/missing_info_page.html (hunk 4 above)
Comment (idisblueflash) [reply data/spiders/rips/missing_info_page.html:2]: I think the HTML file is a good fixture for Unit Tests.

File: scraping/spiders/poland_rjps.py (hunk 6 above)
Comment (idisblueflash) [reply scraping/spiders/poland_rjps.py]: https://github.com/littlepea/safeway-data/pull/18/commits/0a48f991ef028a8da748bd9c6c31cd4003080235

File: data/spiders/rips/addiction.json (hunk 3 above)
Comment (idisblueflash) [reply data/spiders/rips/addiction.json]: https://github.com/littlepea/safeway-data/pull/18/commits/0a48f991ef028a8da748bd9c6c31cd4003080235

File: scraping/spiders/poland_rjps.py (hunk 5 above)
Comment (idisblueflash) [reply scraping/spiders/poland_rjps.py]: https://github.com/littlepea/safeway-data/pull/18/commits/1bc78840051c7e5d348e610d3b954a516c53fc0a

File: data/spiders/rips/missing_info_page.html (hunk 4 above)
Comment (littlepea) [reply data/spiders/rips/missing_info_page.html:2]: Oh, I see, if it's for tests it should be somewhere under the `tests` directory

**Review by littlepea — APPROVED** [review]

**Pull request merged**